network/discovery/shared_conn.go

```diff
@@ -12,7 +12,7 @@ import (
 // messages that were found unprocessable and sent to the unhandled channel by the primary listener.
 // It's copied from https://github.com/ethereum/go-ethereum/blob/v1.14.8/p2p/server.go#L435
 type SharedUDPConn struct {
-	UDPConn   *net.UDPConn // not using embedding to make sure go-ethereum doesn't use unwrapped net.UDPConn's methods
+	*net.UDPConn
```

Comment:
because this change is not present in current `main` release (v2.0.1). And we don't want to introduce more changes to the scope of 2.0.2. 
this can go into next version